pkg/snapshot/snapshotter/snapshot_lease_update.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapshotter
+
+import (
+	"context"
+	"time"
+
+	"github.com/gardener/etcd-backup-restore/pkg/health/heartbeat"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	logger = logrus.NewEntry(logrus.New()).WithField("actor", "FullSnapLeaseUpdater")
+)
+
+// RenewFullSnapshotLeasePeriodically has a timer and will periodically call FullSnapshotCaseLeaseUpdate to renew the fullsnapshot lease until it is updated or stopped.
+// The timer starts upon snapshotter initialization and is reset after every full snapshot is taken.
+func (ssr *Snapshotter) RenewFullSnapshotLeasePeriodically(fullSnapshotLeaseUpdateRetryInterval time.Duration) {
+	ssr.FullSnapshotLeaseUpdateTimer = time.NewTimer(fullSnapshotLeaseUpdateRetryInterval)
+	fullSnapshotLeaseUpdateCtx, fullSnapshotLeaseUpdateCancel := context.WithCancel(context.TODO())
+	defer fullSnapshotLeaseUpdateCancel()
+	for {
```

Comment:
add a log just before starting a for loop

Reply:
Sure.